torch/utils/data/datapipes/iter/sampler.py

```diff
@@ -1,9 +1,46 @@
+import warnings
+
 from torch.utils.data import IterDataPipe, Sampler, SequentialSampler
-from typing import TypeVar, Type, Iterator, Sized
+from typing import Callable, TypeVar, Type, Iterator, Sized
 
 T_co = TypeVar('T_co', covariant=True)
 
 
+class FilterIterDataPipe(IterDataPipe[T_co]):
+    r""" :class:`FilterIterDataPipe`.
+
+    Iterable DataPipe to filter elements from datapipe according to filter_fn.
+    args:
+        datapipe: Iterable DataPipe being filterd
+        filter_fn: Customized function mapping an element to a boolean.
+    """
+    datapipe: IterDataPipe
+    filter_fn: Callable[..., bool]
+
+    def __init__(self,
+                 datapipe: IterDataPipe[T_co],
+                 *args,
+                 filter_fn: Callable[..., bool],
+                 **kwargs,
+                 ) -> None:
+        super().__init__()
+        self.datapipe = datapipe
+        if filter_fn.__name__ == '<lambda>':
```

Comment:
nit: It seems this is a common check for all datapipes that have function inputs. Maybe we can put a generic function somewhere, so we don't need to dup the code for this check.

Reply:
Great point! Will update it using Callable as Base class.